### PR TITLE
set default to True for shuffle in DataLayerNM

### PR DESCRIPTION
**Pull request by ekmb**
Signed-off-by: Evelina Bakhturina <ebakhturina@nvidia.com>

### CODE REVIEW

**Comment by tkornuta-nvidia** [comment]
I totally support that change. PR approved! ;)

**Comment by okuchaiev** [comment]
What if the user downloads a pretrained model, then runs inference and measures, say BLEU score?
They will get 0 and be confused. Happened more than once for OS2S

**Comment by okuchaiev** [comment]
I would keep it False by default. Maybe in train mode we can raise a warning if it is still False

**Comment by tkornuta-nvidia** [comment]
wait... you know that the previous default for shuffle was in fact **true**, right?

**Comment by okuchaiev** [comment]
I think that by default, the DataLayerNM neural module (distributed mode or not) should not shuffle the data. I gave one reason above for why that should be so.  
Pytorch's default behavior is to not shuffle the data https://pytorch.org/docs/stable/data.html?highlight=torch%20utils%20data%20dataloader#torch.utils.data.DataLoader  so I'd like us to be consistent for Pytorch users in addition to the reason I gave above.

Yes, that means that for training, the user will almost always need to set shuffle=True themselves (and we can raise a warning if they don't). 

If setting default shuffle=False in DataLayerNM breaks things - open corresponding issues and let's fix them.
